### ISSUE TITLE
crash in onefile.py:  assert not os.path.exists(source_dir)

### ISSUE THREAD

**Issue by kmesdo** [Nuitka/Nuitka#2885]
Hi,

On Windows 10, I've recently upgraded from python 3.7.4 / Nuitka 2.2.1
to python 3.11.9 / Nuitka 2.2.3.

I'm recompiling one of my projects with the new versions, 
but I get a compile error right at the end:
...
Nuitka-Onefile: Using compression for onefile payload.
Nuitka-Onefile: Onefile payload compression ratio (26.23%) size 67447222 to 17691695.
Nuitka-Onefile: Removing onefile build directory 'MediaPlayerTexture.onefile-build'.
Traceback (most recent call last):
  File "C:\Users\Orac\AppData\Local\Programs\Python\Python311\Lib\site-packages\nuitka\__main__.py", line 161, in <module>
    main()
  File "C:\Users\Orac\AppData\Local\Programs\Python\Python311\Lib\site-packages\nuitka\__main__.py", line 147, in main
    MainControl.main()
  File "C:\Users\Orac\AppData\Local\Programs\Python\Python311\Lib\site-packages\nuitka\MainControl.py", line 1124, in main
    _main()
  File "C:\Users\Orac\AppData\Local\Programs\Python\Python311\Lib\site-packages\nuitka\MainControl.py", line 1035, in _main
    packDistFolderToOnefile(dist_dir)
  File "C:\Users\Orac\AppData\Local\Programs\Python\Python311\Lib\site-packages\nuitka\freezer\Onefile.py", line 52, in packDistFolderToOnefile
    packDistFolderToOnefileBootstrap(onefile_output_filename, dist_dir)
  File "C:\Users\Orac\AppData\Local\Programs\Python\Python311\Lib\site-packages\nuitka\freezer\Onefile.py", line 261, in packDistFolderToOnefileBootstrap
    assert not os.path.exists(source_dir)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
Nuitka-Reports: Compilation crash report written to file 'nuitka-crash-report.xml'. Please include it in your bug report.

I use the following command:

python -m nuitka --onefile --remove-output --windows-disable-console --onefile-tempdir-spec="{TEMP}/_MediaPlayer" --windows-icon-from-ico="MediaPlayer.ico" --output-filename="MediaPlayer" MediaPlayerTexture.py

After it crashes, it leaves behind dir 'MediaPlayerTexture.dist', 
and interestingly, if I run the .exe inside that dir, the app runs just fine.
The packaging into Onefile seems to be the only problem.

Running python MediaPlayerTexture.py from the console works fine.

The older python 3.7.4 / Nuitka 2.2.1 combo works fine too.

**Comment by kayhayen** [comment]
It's asserting that it is able to delete the pre-existing build folders before building them. That not working, my best guess, is that AV tools hold locks while deleting the folder. It is doing 

`        removeDirectory(path=source_dir, ignore_errors=False)` right before, and that it does not work is very strange. This should be the `.onefile-build` folder. What is left behind there? Usually, the exception should be caught, and a retry be made, and if that still arises, it's passed to the outside. 

Our error handler retrying, doesn't necessarily make things continue after succeeding in a local retry though. I don't know if this function changed in Python3.11, but the error handler is called, but the action is not resumed, the error handler should always raise I guess, or not be there at all, it might not be helping. I will therefore switch to a mode, where it now records the function that failed, and retries it for as long as something else fails.

I got a new implementation of it, that I want you to try on the factory branch.

**Comment by kayhayen** [comment]
Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback on whether this is working is very welcome. Please do not share plans to do it; only confirm or deny that it is working.

**Comment by kmesdo** [comment]
Thanks for the feedback. Before going to the factory branch, I tried 
another compile to re-verify the error. The weird thing is that it 
now compiles cleanly (with the new python/nuitka versions)!

Not sure why. The only change I made to my environment was for a 
different project which required me to install the following packages:

   pip install requests
   pip install ratelimit
   pip install PyJWT

I'm not sure that there is any value in trying the factory branch now, 
but if I get a similar error in a different project, I will give it a try.

Thanks for your help.
